src/bit_manipulation/basic.rs

```diff
@@ -75,38 +502,239 @@ pub fn count_ones(bits: i8) -> i8 {
+///
+/// Returns the number of equal bits between `a` and `b`.
+///
+/// see: [Hamming Distance](https://en.wikipedia.org/wiki/Hamming_distance)
+///
+/// This is the inverse of the Hamming Distance.
+///
+/// note that the sign bit is not considered in this operation.
+///
+/// # Arguments
+///
+/// `a` - The first number to compare.
+/// `b` - The second number to compare.
+///
+/// # Returns
+///
+/// The number of equal bits between `a` and `b`.
+///
+/// # Panic
+///
+/// This function will not panic.
+///
+/// # Examples
+///
+/// ```rust
+/// use rust_algorithms::bit_manipulation::bit_equivalence;
+///
+/// assert_eq!(0, bit_equivalence(0b000_0000, 0b111_1111));
+/// assert_eq!(6, bit_equivalence(0b000_0001, 0b000_0000));
+/// assert_eq!(7, bit_equivalence(0b111_1111, 0b111_1111));
+///
+/// ```
+///
 pub fn bit_equivalence(a: i8, b: i8) -> i8 {
     count_ones(!(a ^ b))
 }
 
+/// Calculates the bit distance between two numbers.
+///
+/// Returns the number of different bits between `a` and `b`.
+///
+/// see: [Hamming Distance](https://en.wikipedia.org/wiki/Hamming_distance)
+///
+/// note that the sign bit is not considered in this operation.
+///
+/// # Arguments
+///
+/// `a` - The first number to compare.
+/// `b` - The second number to compare.
+///
+/// # Returns
+///
+/// The number of different bits between `a` and `b`.
+///
+/// # Panic
+///
+/// This function will not panic.
+///
+/// # Examples
+///
+/// ```rust
+/// use rust_algorithms::bit_manipulation::bit_distance;
+///
+/// assert_eq!(0, bit_distance(0, 0));
+/// assert_eq!(0, bit_distance(127, 127));
+/// assert_eq!(0, bit_distance(55, 55));
+/// assert_eq!(6, bit_distance(0b111_1111, 0b000_0100));
+/// assert_eq!(6, bit_distance(0b101_1011, 0b000_0100));
+/// assert_eq!(7, bit_distance(0b111_1111, 0b000_0000));
+///
+/// ```
+///
 pub fn bit_distance(a: i8, b: i8) -> i8 {
     count_ones(a ^ b)
 }
 
-// `x & (x - 1) == 0`
+
+/// Checks if a number is a power of two.
+///
+/// Returns true if `bits` is a power of two, otherwise false.
+///
+/// see: [Power of Two](https://en.wikipedia.org/wiki/Power_of_two)
+///
+/// This function uses the following bitwise operation:
+///
+/// `x & (x - 1) == 0`
+///
+/// # Arguments
+///
+/// `bits` - The number to check.
+///
+/// # Returns
+///
+/// True if `bits` is a power of two, otherwise false.
+///
+/// # Panic
+///
+/// This function will not panic.
+///
+/// # Examples
+///
+/// ```rust
+/// use rust_algorithms::bit_manipulation::is_power_of_two;
+///
+/// assert!(is_power_of_two(0));
+/// assert!(is_power_of_two(2));
+/// assert!(is_power_of_two(16));
+/// assert!(is_power_of_two(64));
+///
+/// assert!(!is_power_of_two(33));
+/// assert!(!is_power_of_two(124));
+/// assert!(!is_power_of_two(-13));
+///
+/// ```
+///
 pub fn is_power_of_two(bits: i8) -> bool {
     bits & (bits.wrapping_sub(1)) == 0
 }
 
 // `((x | (x - 1)) + 1) & x == 0`
 //
 // Also `((x & -x) + x) & x == 0`
+
```

Comment:
Hmmm, it would be weird to leave these comments here, right?

So, it wasn't really clear here, but this function doesn't check if a number is a power of two, but instead if it is *the difference* of two powers of two.

Namely `2^k - 2^j`.

If it's not clear lmk and I'll expand a bit more on this.